### PR TITLE
remove outdated v2 source code and tests from v3 branch

### LINKED ISSUE #1849: [v3] remove old v2 code entirely
This is a more broad alternative to #1791 -- basically we remove all the old v2 code in favor of the v3 efforts. Happy to hear thoughts on this.

### PR DESCRIPTION
**Pull request by jhamman**
This PR removes the `zarr.v2` namespace from the v3 branch. I'm moving forward with this based on discussions in #1849 and in the weekly developer meeting. I know there are remain varying views on the best way to minimize disruption for existing users of zarr 2.x so I'll leave this up for at least a week to give folks a chance to consider the consequences of this PR. 

closes #1849

cc @zarr-developers/python-core-devs 

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [x] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

File: tests/v3/test_codecs/test_codecs.py [hunk 1]
```diff
@@ -255,90 +238,6 @@ async def test_delete_empty_chunks(store: Store) -> None:
     assert await store.get(f"{path}/c0/0", prototype=default_buffer_prototype()) is None
 
 
-@pytest.mark.parametrize("store", ("local", "memory"), indirect=["store"])
-async def test_zarr_compat(store: Store) -> None:
```
Comment (jhamman) [inline tests/v3/test_codecs/test_codecs.py:259]: drawing attention to this change which we added to develop confidence that we were doing backward compatible things. 

We may consider replacing this with a new CI workflow / test suite that generates data w/ 2.18 and consumes it with the v3 branch.

**Review by normanrz — APPROVED** [review]
Removing this is a major milestone in our prep for the v3 release. I think it is important to remove it to reduce confusion and to let the community help us identify compatibility issues that we still need to address.
Thanks @jhamman!

**Comment by d-v-b** [comment]
looks great, thanks joe! 🚀

File: tests/v3/test_codecs/test_codecs.py (hunk 1 above)
Comment (dstansby) [reply tests/v3/test_codecs/test_codecs.py:259]: Is it worth opening an issue to keep track of replacing this?

Comment (jhamman) [reply tests/v3/test_codecs/test_codecs.py:259]: https://github.com/zarr-developers/zarr-python/issues/2185

**Review by rabernat — APPROVED** [review]
Onward and upward! :rocket:

**Pull request merged**